packages/engine/src/xr/functions/WebXRFunctions.ts

```diff
@@ -147,15 +133,11 @@ export const startWebXR = (): void => {
 
   // Map input sources
   // Sometimes the input sources are not available immidiately
-  setTimeout(() => {
-    mapXRControllers(inputData)
-    proxifyXRInputs(world.localClientEntity, inputData)
-  }, 1000)
-
-  addComponent(world.localClientEntity, XRInputSourceComponent, inputData)
-
-  bindXRHandEvents()
+  await delay(1000)
```

Comment:
can we use a threejs or WebXR API event instead of a delay?

Reply:
For some reason, the 'connect' event from three.js does not trigger the first time going into immersive mode. That's why I have used delay. I'll try to find an alternative solution.

Reply:
Replaced the delay with 'inputsourceschange' event from XRSession. Testing on the emulator and Quest 2 seem to be working without issues.